### ISSUE TITLE
Question on Label Format: Individual Actions vs. Grouped Action Sequences

### ISSUE THREAD

**Issue by hbbOooO** [hwanyu112/Latent-Sketchpad#3]
Hi @hwanyu112,Thank you for your outstanding work on this project! 

I have a question regarding the label format for the training data.

I noticed that in the publicly available test data, each action is treated as an individual entry in the label_actions list. However, the paper mentions that "Each state comprises a short sequence of $k \in [4, 6]$ actions."

Based on this description, I prepared my training data using a state-by-state reasoning approach, where actions are grouped into sequences within the <actions> tags and the label_actions list (as shown in my example below).

My Question:Does this grouped, state-based format align with your training requirements, or should the model expect individual actions as seen in the test set?

I would appreciate your guidance on which format is preferred for optimal performance.

**Comment by hbbOooO** [comment]
Example:

```
"input_text": "Given the maze in the input image <image>, determine a valid action sequence to navigate from the starting point (green arrow) to the endpoint (red circle).The black lines represent walls, and the white areas are traversable paths. Each action in the sequence must be one of the following: \"go forward\": Move straight until reaching the next turn or intersection. \"turn left\": Rotate left before moving forward. \"turn right\": Rotate right before moving forward. During the reasoning process, clearly mark each confirmed action using the format  <actions>confirmed action</actions>.\n",



"label_text": "Now, let's reason through the next set of steps. Observing the current state and the path ahead, I need to execute a sequence of moves to reach the next key state. The sequence of actions is: move forward along the path, then turn right at the junction, then move forward along the path, then turn right at the junction. These actions will navigate through the intermediate path segments. The actions of this part are <actions>go forward, turn right, go forward, turn right</actions>\n<image>\nLet's continue to the next state.\nNow, let's reason through the next set of steps. Observing the current state and the path ahead, I need to execute a sequence of moves to reach the next key state. The sequence of actions is: move forward along the path, then turn left at the junction, then move forward along the path, then turn left at the junction, then move forward along the path, then turn right at the junction. These actions will navigate through the intermediate path segments. The actions of this part are <actions>go forward, turn left, go forward, turn left, go forward, turn right</actions>\n<image>\nLet's continue to the next state.\nNow, let's reason through the next set of steps. Observing the current state and the path ahead, I need to execute a sequence of moves to reach the next key state. The sequence of actions is: move forward along the path, then turn right at the junction, then move forward along the path, then turn right at the junction, then move forward along the path. These actions will navigate through the intermediate path segments. The actions of this part are <actions>go forward, turn right, go forward, turn right, go forward</actions>\n<image>\nThe inference process is complete.",



"label_actions": [

    "go forward, turn right, go forward, turn right",

    "go forward, turn left, go forward, turn left, go forward, turn right",

    "go forward, turn right, go forward, turn right, go forward"

],
```

**Comment by hwanyu112** [comment]
Hi, thank you for your interest in our project.

Regarding your example, the grouped, state-based format is conceptually close to the reasoning process described in the paper, but in our actual training pipeline we further decompose the label_text into multiple states. 

More specifically, for an example like the one you shared, we would split it into 3 individual training samples, one for each state. This is mainly to ensure that, under our current pipeline implementation, only one image is involved in the loss computation for each sample. In addition, current [dataset ](https://github.com/hwanyu112/Latent-Sketchpad/blob/418062f166d79d3d3d9b28e9324731f090739576/data/dataset.py#L195) will automatically add the <eos> token only to the last sample after this decomposition.

As for label_actions, in our training setup each string contains only a single action, consistent with the format used in the test set. So instead of grouping multiple actions into one entry, we represent them as individual actions.

To make this clearer, I will upload a concrete training data example shortly for reference.
